src/Symfony/Component/DependencyInjection/Tests/Compiler/ResolveBindingsPassTest.php

```diff
@@ -129,4 +130,20 @@ public function testWithNonExistingSetterAndBinding()
         $pass = new ResolveBindingsPass();
         $pass->process($container);
     }
+
+    public function testSyntheticServiceWithBind()
+    {
+        $container = new ContainerBuilder();
+
+        $container->register('synthetic.service')->setSynthetic(true)->setPublic(true)->setBindings(['$foo' => 'bar']);
+        $pass = new ResolveBindingsPass();
+
+        try {
+            $pass->process($container);
+        } catch (InvalidArgumentException $e) {
```

Comment:
I guess you rather want to use `expectException()` here:

```php
$this->expectException(InvalidArgumentException::class);
$this->expectExceptionMessageRegexp('/Invalid service "synthetic.service": class "" does not exist./');

$pass->process($container);
```

Reply:
no, i need to check if exception does not contain that message because invalid argument is thrown for different reason anyway

Reply:
is there another way? this looks very fragile to me, e.g. if we change the error message, a regression can happen...

Reply:
i couldn't think of any other way. I can't just catch InvalidArgumentException because it is thrown anyways for different reason but I need to check if exception doesn't have this specific part in it

Reply:
I managed to improve the test case while merging.